Predicting Diabetes Status - Pima Indian Diabetes data

In [ ]:
import pandas as pd
import numpy as np

In [ ]:
pima = pd.read_csv('/content/diabetes (1).csv')
pima.head()

Pregnancies  Glucose  BloodPressure  ...  DiabetesPedigreeFunction  Age  Outcome
0            6      148             72  ...                     0.627   50        1
1            1       85             66  ...                     0.351   31        0
2            8      183             64  ...                     0.672   32        1
3            1       89             66  ...                     0.167   21        0
4            0      137             40  ...                     2.288   33        1

[5 rows x 9 columns]

In [ ]:
pima.count()

Pregnancies                 768
Glucose                     768
BloodPressure               768
SkinThickness               768
Insulin                     768
BMI                         768
DiabetesPedigreeFunction    768
Age                         768
Outcome                     768
dtype: int64

In [ ]:
pima.describe()

Pregnancies     Glucose  ...         Age     Outcome
count   768.000000  768.000000  ...  768.000000  768.000000
mean      3.845052  120.894531  ...   33.240885    0.348958
std       3.369578   31.972618  ...   11.760232    0.476951
min       0.000000    0.000000  ...   21.000000    0.000000
25%       1.000000   99.000000  ...   24.000000    0.000000
50%       3.000000  117.000000  ...   29.000000    0.000000
75%       6.000000  140.250000  ...   41.000000    1.000000
max      17.000000  199.000000  ...   81.000000    1.000000

[8 rows x 9 columns]

In [ ]:
feature =['Pregnancies', 'Insulin', 'BMI', 'Age']

In [ ]:
x = pima[feature]
y = pima.Outcome

In [ ]:
from sklearn.model_selection import train_test_split

In [ ]:
x_train, x_test, y_train, y_test = train_test_split(x, y, random_state = 0)

In [ ]:
from sklearn.linear_model import LogisticRegression

In [ ]:
log_reg  = LogisticRegression()
log_reg.fit(x_train, y_train)

LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=100,
                   multi_class='auto', n_jobs=None, penalty='l2',
                   random_state=None, solver='lbfgs', tol=0.0001, verbose=0,
                   warm_start=False)

In [ ]:
y_pred = log_reg.predict(x_test)

In [ ]:
from sklearn import metrics

In [ ]:
print(metrics.accuracy_score(y_test,y_pred))

0.6770833333333334


In [ ]:
y_test.value_counts()

0    130
1     62
Name: Outcome, dtype: int64

In [ ]:
y_test.mean()

0.3229166666666667

In [ ]:
print("true diabetes label: ", y_test.values[0:15])
print("Predicted Labels:", y_pred[0:15])

true diabetes label:  [1 0 0 1 0 0 1 1 0 0 1 1 0 0 0]
Predicted Labels: [0 0 0 0 0 0 0 1 0 1 0 1 0 0 0]


In [ ]:
print(metrics.confusion_matrix(y_test, y_pred))

[[114  16]
 [ 46  16]]


114 - true positive - predicted that people 
have diabeties    
16 - false positive - errors -                             
46 - true negative - error                                        
16 - false negative- predicted that people donot have diabeties.

error = (FP + FN)/(TP+TN+FP+FN)                                                 sensitivity is calculated using recall score: it is TP/TP+FP   -- it gives how often is it correct.

In [ ]:
print(metrics.recall_score(y_test, y_pred))

0.25806451612903225


In [ ]:
print(metrics.precision_score(y_test, y_pred))

0.5
